In [1]:
import pandas as pd
from selenium import webdriver 
from bs4 import BeautifulSoup
import time
import re

In [2]:
from datetime import date, timedelta, datetime
import os.path

In [3]:
t0 =  date.today()

In [4]:
news_url='http://www.99huizhou.com/forum.php?mod=forumdisplay&fid=78&filter=typeid&typeid=162'

datetime_str = t0
fname1 = './data/general/temp/' + '99huizhou-huiwenhua' + '.csv'
fname2 = './data/general/' + '99huizhou-huiwenhua' + '.csv'


In [5]:
def get_columns():
    return ('dt', 'source', 'domain', 'url', 'title', 'summary', 'image', 'brief', 'body', 'raw_page', 'from') 
def get_empty_columns():
    return ('', '', '', '', '', '', '', '', '', '', '') 
def get_empty_columns1():
    return ('', '', '', '', '', '', '', '', '', '', '','') 



In [6]:
browser = webdriver.Firefox() 
#browser = webdriver.Chrome('/Users/nhu2000/Downloads/Chromedriver')

In [12]:
if os.path.isfile(fname1): 
    df = pd.read_csv(fname1)
    idx = len(df.index)
else:
    df = pd.DataFrame(columns = get_columns())
    idx = 0

In [13]:
if os.path.isfile(fname2): 
    df_check = pd.read_csv(fname2)
    idx = len(df.index)
else:
    df_check = pd.DataFrame(columns = get_columns())
    idx = 0

if os.path.isfile(fname2): 
    df_0 = pd.read_csv(fname2)
else:
    df_0 = pd.DataFrame(columns = get_columns())

In [14]:
print len(df_check)

0


In [15]:
print len(df)

0


browser.get(news_url)
time.sleep(5)

In [57]:
url_0 = news_url
print url_0
browser.get(url_0)
time.sleep(5)

http://www.99huizhou.com/forum.php?mod=forumdisplay&fid=78&filter=typeid&typeid=162


In [108]:
idx = 0
not_existed = True
df = pd.DataFrame(columns = get_columns())

In [111]:
soup = BeautifulSoup(browser.page_source)
#sub_text = soup.findAll('tr', {'class':'eis_bd'})
#sub_text = soup.findAll('tr', {'class':'fl_row'})
#sub_text = soup.findAll('div', {'id':'threadlist'})
sub_text = soup.findAll('tbody', {'id': lambda L: L and L.startswith('normalthread_')})

for item in sub_text:
    #print 'www.hzwh.com/' + item.a['href']
    #print t1.text
    #print item.text

    t1 = item.find('a', {'class':'s xst'})    
    t2 = item.find('td', {'class':'by'})    
    #t0 = item.find('td', {'class':'eis_fl_forum'})    
    #if t0 != None:
    #    t1 = t0.h2.a
    #print t1.text
    #print t1['href']
    #print t2.em.text
    
    if not_existed == True and t1 != None:   
        print "-------- add", 
        print t1.text
        #print t1['href']
        df.loc[idx] = get_empty_columns()        
        df.loc[idx]['url'] = 'http://www.99huizhou.com/' + t1['href']
        df.loc[idx]['title'] = t1.text
        df.loc[idx]['source'] = '故园徽州'
        df.loc[idx]['dt'] = t2.em.text
        idx += 1

-------- add 谁说15号的活动没有意义,
-------- add 德国建筑专家在樘樾
-------- add 史墅老匾
-------- add 宏村承志堂的木雕的艺术为什么没有隔壁的艺术高
-------- add 德国建筑专家在木坑竹海
-------- add 德国建筑专家在宏村
-------- add 张脉贤在<中国龙川 胡氏宗祠木雕解读>一书中的序言
-------- add 德国建筑专家参观徽博录象1
-------- add 隆重推出徽州木雕的一本好书《中国龙川　胡氏宗祠木雕解读》
-------- add 这是什么玩意?
-------- add 徽州三雕群是21434021欢乐你的加入
-------- add 老外眼中的徽州雕刻
-------- add 徽州建筑木雕什么时候出现雕刻人物
-------- add 徽州木雕的标志性是什么？
-------- add 《徽州明清雀替读解》一书木雕图片脚本
-------- add 徽州九狮一对
-------- add 【喜讯】祝贺老土徽州三雕版今日开盘！


In [112]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 67 entries, 0 to 66
Data columns (total 11 columns):
dt          67 non-null object
source      67 non-null object
domain      67 non-null object
url         67 non-null object
title       67 non-null object
summary     67 non-null object
image       67 non-null object
brief       67 non-null object
body        67 non-null object
raw_page    67 non-null object
from        67 non-null object
dtypes: object(11)
memory usage: 6.3+ KB


In [56]:
df.head()

dt source domain  \
0    2017-2-9   故园徽州          
1    2017-1-1   故园徽州          
2  2016-11-23   故园徽州          
3  2016-10-25   故园徽州          
4  2016-10-17   故园徽州          

                                                 url  \
0  forum.php?mod=viewthread&tid=195025&extra=page...   
1  forum.php?mod=viewthread&tid=194254&extra=page...   
2  forum.php?mod=viewthread&tid=192264&extra=page...   
3  forum.php?mod=viewthread&tid=189658&extra=page...   
4  forum.php?mod=viewthread&tid=189555&extra=page...   

                        title summary image brief body raw_page from  
0  【春节民俗活动预告】——中云镇桃溪坑头“抬汪公”民俗                                         
1               2017.01.01的早上                                         
2                       上学去喽！                                         
3        流淌的登源河（29）：老造反成了普通看客                                         
4                    沙溪“淩”姓来由

In [113]:
if(len(df) != 0):
    df.to_csv('../data/topics/100/110/99huizhou.csv', encoding='utf-8', index=False)
#    df.to_csv('../data/topics/400/420/99huizhou.csv', encoding='utf-8', index=False)

In [118]:
for i in range(0, len(df)):
    url = 'http://www.99huizhou.com/' + df.loc[i]['url']
    browser.get(url)
    time.sleep(15)
    
    soup = BeautifulSoup(browser.page_source)
    sub_text = soup.findAll('div', {'class':'t_fsz'})
    
    fst = 0
    for item in sub_text:
        if fst == 0:
            #print item.text
            print i, df.loc[i]['title']
            #print 'add --', item.text
            df.loc[i]['body'] = item.text
            df.loc[i]['raw_page'] = browser.page_source    
            fst = 1


0 【春节民俗活动预告】——中云镇桃溪坑头“抬汪公”民俗
1 2017.01.01的早上
2 上学去喽！
3 流淌的登源河（29）：老造反成了普通看客
4 沙溪“淩”姓来由
5 流淌的登源河（25）：一月风暴刮到绩溪
6 流淌的登源河（24）：文革号角悄然吹响
7 流淌的登源河（22）：和苏村的亲密接触
8 流淌的登源河（20）：反右倾是第一位的
9 流淌的登源河（18）：农民的软抗高招
10 流淌的登源河（16）：龙丛源里的农家
11 流淌的登源河（14）：十三号公寓
12 流淌的登源河（12）：饭票和钞票的困扰
13 流淌的登源河（10）：头一回当工作组
14 流淌的登源河（6）
15 流淌的登源河（8）
16 流淌的登源河（4）
17 流淌的登源河（2）
18 流淌的登源河（寄语与目录）
19 养生福地，美食天堂——绩溪
20 母亲们节日快乐
21 阳春三月祭汪王（3月31日修改版）
22 阳春三月祭汪王
23 丙申清明汪王祭活动公告
24 开门红
25 绩溪文承堂章氏家训展馆
26 《徽州往事》再登国家大剧院
27 故园旗帜在婺源飘扬
28 婺源：“傩娃”踩春传承非遗 人花相融天人合一
29 徽州剪纸
30 上花轿
31 两本剪纸
32 茶仙子鲍丽丽在祁门演绎一场关于红茶的饕餮盛宴
33 新安程氏乙未年清明篁墩公祭
34 歙县南乡民间的两篇“汪公大帝祭文”
35 请徽友们关注：3月29日   清明汪王祭    首届中国黄山汪华文化节
36 迄今描述"歙县蜈蚣岭村"最为精彩的一篇文章，家乡人请转起
37 解板
38 羊年   汪华文化短讯五则
39 绩溪余川将举办"汪王祭祀和抬汪公"大型民俗活动
40 祁门县文堂村举行乡约家法宣讲仪式
41 追寻徽州的淳朴印迹
42 过年习俗完整版
43 【历史上的今天】十八年前的“6·30”洪水
44 绩溪民间禁忌趣谈
45 “老家的味道”在歙县拍摄 央视7套小年播出
46 邑小士多,名儒辈出——徽州绩溪
47 狮子
48 好文阅读 斜阳残照徽州梦
49 戏曲表演乐市民
50 捉知了_-----高高的窜天杨
51 齐云山进香记
52 大姐头发有点长
53 祠堂的血与泪
54 前两个字什么意思
55 历史传奇故事： 苏小妹 三 难  新 郎
56 未见屯溪面十里闻茶香.........忆屯溪古缜茶季
57 俺滴师弟又打篮了
5

In [107]:
print url

forum.php?mod=viewthread&tid=195025&extra=page%3D1%26filter%3Dtypeid%26typeid%3D162


In [120]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 899 entries, 0 to 898
Data columns (total 11 columns):
dt          899 non-null object
source      899 non-null object
domain      899 non-null object
url         899 non-null object
title       899 non-null object
summary     899 non-null object
image       899 non-null object
brief       899 non-null object
body        899 non-null object
raw_page    899 non-null object
from        899 non-null object
dtypes: object(11)
memory usage: 84.3+ KB


In [12]:
for i in range(0, 1):
    url_0 = news_url
    print url_0
    browser.get(url_0)
    time.sleep(5)

    soup = BeautifulSoup(browser.page_source)
    sub_text = soup.findAll('div', {'class':'listA_title'})

    for item in sub_text:
        t1 = item.find('div', {'class':'listA_Time'})
        #print 'www.hzwh.com/' + item.a['href']
        #print t1.text
        print item.a.text
        url = 'www.hzwh.com/' + item.a['href']
        
        not_existed = True
        #check if in cleaned set
        if t1 != None and len(df_check) != 0:
            t0 = abs(sum(df_check['url'].str.find(url))) 
            if t0 != len(df_check):
                not_existed = False

        #check if in dynamic set
        if t1 != None and len(df) != 0 and not_existed == True:
            t0 = abs(sum(df['url'].str.find(str(url)))) 
            if t0 != len(df):
                not_existed = False

        #print t0, len(df), t1

        if not_existed == True and t1 != None:   
            print "-------- add"
            df.loc[idx] = get_empty_columns()        
            df.loc[idx]['url'] = url
            df.loc[idx]['title'] = item.a.text
            df.loc[idx]['source'] = '徽州文化网'
            df.loc[idx]['dt'] = t1.text
            idx += 1

http://www.hzwh.com/toplistb.asp?id=229&page=0
传统徽州仁商义贾践行商君子之道的历史启示
《阅读徽州》第七编“徽州科技”
传统徽商的诚信为本和仁义向善
传统徽商的诚信为本和仁义向善
黟县古城改造重要的是务须保住文教根脉
推动徽州传统文化的普及宣传
徽学与当代学术 
“儒学与地域文化：徽学国际学术研讨会”综述
洪经纶墓葬地点的疏证
徽州乡镇志中所见明清民间救荒措施
徽州文化全书
论文化生产力与中部地区的经济发展
徽州书院的讲会
明代海上徽帮与宁波海上丝绸之路
徽州昌溪周氏家族与宁波近代商业发展史
徽商历史的文化价值批判 
徽商
徽商文化及特点
婺源徽剧音乐研究
古徽州和谐乡村社会建设的经验


/usr/local/lib/python2.7/site-packages/bs4/__init__.py:181: UserWarning: No parser was explicitly specified, so I'm using the best available HTML parser for this system ("html5lib"). This usually isn't a problem, but if you run this code on another system, or in a different virtual environment, it may use a different parser and behave differently.

The code that caused this warning is on line 162 of the file /usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py. To get rid of this warning, change code that looks like this:

 BeautifulSoup([your markup])

to this:

 BeautifulSoup([your markup], "html5lib")

  markup_type=markup_type))


In [13]:
print len(df)

0


In [12]:
df.head(50)

dt source domain                              url  \
0   2004-03-03  徽州文化网         www.hzwh.com/article.asp?id=533   
1   2004-03-03  徽州文化网         www.hzwh.com/article.asp?id=532   
2   2004-03-09  徽州文化网         www.hzwh.com/article.asp?id=530   
3   2004-03-22  徽州文化网         www.hzwh.com/article.asp?id=528   
4   2004-03-31  徽州文化网         www.hzwh.com/article.asp?id=527   
5   2004-04-02  徽州文化网         www.hzwh.com/article.asp?id=525   
6   2004-04-11  徽州文化网         www.hzwh.com/article.asp?id=523   
7   2004-04-12  徽州文化网         www.hzwh.com/article.asp?id=521   
8   2004-04-15  徽州文化网         www.hzwh.com/article.asp?id=520   
9   2004-04-15  徽州文化网         www.hzwh.com/article.asp?id=518   
10  2004-05-11  徽州文化网         www.hzwh.com/article.asp?id=517   
11  2004-05-13  徽州文化网         www.hzwh.com/article.asp?id=516   
12  2009-04-28  徽州文化网         www.hzwh.com/article.asp?id=505   

                  title summary image brief body raw_page from  
0            旧时徽州税票名称种种                                         
1           小小大阜村 爷孙宰相俩                                         
2            李良春与《惯痴文集》                                         
3                 古老的石林                                         
4                 洪钧行书联                                         
5            万安“中国罗盘之乡”                                         
6                   吕仙宫                                         
7               溪 头 探 古                                         
8        五绝《小桃源》中否李白所作？                                         
9               解读戴震诗一首                                         
10        安徽休宁——中国第一状元县                                         
11           独树一帜的徽州税文化                                         
12  扬弃古徽州廉政文化 促进新黄山廉政建设

In [28]:
print len(df)

0


In [163]:
def url_get(url):
    browser.get(url)
    print url
    time.sleep(5)
    return browser.page_source

In [164]:
df['dt'] = df_check['dt']
df['source'] = df_check['source']
df['url'] = df_check['url']
df['title'] = df_check['title']
df['raw_page'] = df_check.url.map(lambda x: url_get('http://' + x) )  

http://www.hzwh.com/article.asp?id=2538
http://www.hzwh.com/article.asp?id=2537
http://www.hzwh.com/article.asp?id=2536
http://www.hzwh.com/article.asp?id=2535
http://www.hzwh.com/article.asp?id=2534
http://www.hzwh.com/article.asp?id=2533
http://www.hzwh.com/article.asp?id=2099
http://www.hzwh.com/article.asp?id=2098
http://www.hzwh.com/article.asp?id=1848
http://www.hzwh.com/article.asp?id=1847
http://www.hzwh.com/article.asp?id=1846
http://www.hzwh.com/article.asp?id=1794
http://www.hzwh.com/article.asp?id=1695
http://www.hzwh.com/article.asp?id=1694
http://www.hzwh.com/article.asp?id=1693
http://www.hzwh.com/article.asp?id=1624
http://www.hzwh.com/article.asp?id=1623
http://www.hzwh.com/article.asp?id=1622
http://www.hzwh.com/article.asp?id=1455
http://www.hzwh.com/article.asp?id=1418
http://www.hzwh.com/article.asp?id=1417
http://www.hzwh.com/article.asp?id=1416
http://www.hzwh.com/article.asp?id=1415
http://www.hzwh.com/article.asp?id=1414
http://www.hzwh.com/article.asp?id=1347


TimeoutException: Message: Error loading page, timed out (onDOMContentLoaded)


In [165]:
df['url'] = 'http://' + df['url']

In [119]:
df.head()

dt source domain  \
0    2017-2-9   故园徽州          
1    2017-1-1   故园徽州          
2  2016-11-23   故园徽州          
3  2016-10-25   故园徽州          
4  2016-10-17   故园徽州          

                                                 url  \
0  forum.php?mod=viewthread&tid=195025&extra=page...   
1  forum.php?mod=viewthread&tid=194254&extra=page...   
2  forum.php?mod=viewthread&tid=192264&extra=page...   
3  forum.php?mod=viewthread&tid=189658&extra=page...   
4  forum.php?mod=viewthread&tid=189555&extra=page...   

                        title summary image brief  \
0  【春节民俗活动预告】——中云镇桃溪坑头“抬汪公”民俗                       
1               2017.01.01的早上                       
2                       上学去喽！                       
3        流淌的登源河（29）：老造反成了普通看客                       
4                    沙溪“淩”姓来由                       

                                                body  \
0  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
1  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
2  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
3  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   
4  \n\n\n\n\n马上注册成为会员，享用更多功能，让你轻松玩转徽州\n您需要 登录 才可以...   

                                            raw_page from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...       
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...

In [152]:
characters_num = 40
def get_brief(txt):
    soup = BeautifulSoup(txt)
    sub_text = soup.findAll('div', {'class':'article_contents'})
    for item in sub_text:
        t20 = item.text

    idx = 2
    t30 = t20.split(u'\n')
    
    cnt = 0
    tt = ''
    idx = 0
    for k in range(0, len(t30)):
        if (len(t30[k]) > 40 and len(tt) < 40):
            idx = k
            tt = t30[k].strip()
    
    if len(tt) == 0:
        tt = t20.strip()
                
    #print idx

    t40 = tt[0:characters_num]
    t50 = t40.encode('utf-8') + '..'
    #print idx, t40
    return t50

In [158]:
df['brief'] = df.raw_page.map(lambda x: get_brief(x) )  

In [154]:
def get_body(txt):
    soup = BeautifulSoup(txt)
    sub_text = soup.findAll('div', {'class':'article_contents'})
    for item in sub_text:
        t20 = item.text

    return t20

In [159]:
df['body'] = df.raw_page.map(lambda x: get_body(x) )  

cnt = 0
for item in df['brief']:
    print cnt, item
    cnt += 1

cnt = 0
for item in df['body']:
    print cnt, item
    cnt += 1

soup = BeautifulSoup(df.loc[527]['raw_page'])
sub_text = soup.findAll('div', {'class':'article_contents'})
for item in sub_text:
    t20 = item.text
t30 = t20.split(u'\n')

tt = t20.strip()
print len(t30), tt[0:40], t20

In [161]:
df.head()

dt source domain                               url  \
0  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2538   
1  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2537   
2  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2536   
3  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2535   
4  2016-04-08  徽州文化网    NaN  www.hzwh.com/article.asp?id=2534   

                  title summary image  \
0  传统徽州仁商义贾践行商君子之道的历史启示     NaN   NaN   
1       《阅读徽州》第七编“徽州科技”     NaN   NaN   
2        传统徽商的诚信为本和仁义向善     NaN   NaN   
3        传统徽商的诚信为本和仁义向善     NaN   NaN   
4    黟县古城改造重要的是务须保住文教根脉     NaN   NaN   

                                        brief  \
0  全国人大代表、民盟安徽省委副主委、安徽省文联副主席钱念孙先生在《光明日报》201..   
1  长期以来，得益于灵山秀水自然环境的滋养、兴学重教社会风气的熏染、经世致用治学风格..   
2  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
3  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
4  黟县进行古城改造并申报国家级历史文化名城总体方案正在逐步完善，各项工作正在有条不..   

                                                body  \
0  传统徽州仁商义贾践行商君子之道的历史启示\n郗延红\n全国人大代表、民盟安徽省委副主委、安徽...   
1  《阅读徽州》第七编“徽州科技”（市委市政府通过市社科联组织编写）\n中国科学技术学大出版社出...   
2  传统徽商的诚信为本和仁义向善\n\n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫”...   
3  传统徽商的诚信为本和仁义向善\n \n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫...   
4  黟县古城改造重要的是务须保住文教根脉\n                         郗...   

                                            raw_page from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN  
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...  NaN

In [127]:
df.loc[24]['url']

'www.hzwh.com/article.asp?id=1347'

In [121]:
if(len(df) != 0):
    df.to_csv('../data/complete/hzwh_huixueyuanpu.csv', encoding='utf-8', index=False)

In [114]:
if(browser):
    browser.quit()

In [31]:
for item in df['title']:
    print item

In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 11 columns):
dt          0 non-null object
source      0 non-null object
domain      0 non-null object
url         0 non-null object
title       0 non-null object
summary     0 non-null object
image       0 non-null object
brief       0 non-null object
body        0 non-null object
raw_page    0 non-null object
from        0 non-null object
dtypes: object(11)
memory usage: 0.0+ bytes


In [173]:
df2 = pd.read_csv('../data/complete/hzwh_huixueyuanpu.csv')

In [174]:
df2.head()

dt source  domain                                      url  \
0  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2538   
1  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2537   
2  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2536   
3  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2535   
4  2016-04-08  徽州文化网     NaN  http://www.hzwh.com/article.asp?id=2534   

                  title  summary  image  \
0  传统徽州仁商义贾践行商君子之道的历史启示      NaN    NaN   
1       《阅读徽州》第七编“徽州科技”      NaN    NaN   
2        传统徽商的诚信为本和仁义向善      NaN    NaN   
3        传统徽商的诚信为本和仁义向善      NaN    NaN   
4    黟县古城改造重要的是务须保住文教根脉      NaN    NaN   

                                        brief  \
0  全国人大代表、民盟安徽省委副主委、安徽省文联副主席钱念孙先生在《光明日报》201..   
1  长期以来，得益于灵山秀水自然环境的滋养、兴学重教社会风气的熏染、经世致用治学风格..   
2  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
3  徽州古楹联云“仁中取利真君子/义内求财大丈夫”。这是对传统徽商群体中那些仁商义贾..   
4  黟县进行古城改造并申报国家级历史文化名城总体方案正在逐步完善，各项工作正在有条不..   

                                                body  \
0  传统徽州仁商义贾践行商君子之道的历史启示\n郗延红\n全国人大代表、民盟安徽省委副主委、安徽...   
1  《阅读徽州》第七编“徽州科技”（市委市政府通过市社科联组织编写）\n中国科学技术学大出版社出...   
2  传统徽商的诚信为本和仁义向善\n\n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫”...   
3  传统徽商的诚信为本和仁义向善\n \n郗延红\n徽州古楹联云“仁中取利真君子/义内求财大丈夫...   
4  黟县古城改造重要的是务须保住文教根脉\n                         郗...   

                                            raw_page  from  
0  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
1  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
2  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
3  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN  
4  <html xmlns="http://www.w3.org/1999/xhtml"><he...   NaN